language/types_values_variables.md

```diff
@@ -415,6 +502,237 @@ only, it can not be used in the Puppet Programming Language.
     String ∪ (T ∈ Scalar)  → Scalar
     String ∪ (T ∉ Scalar)  → Any
 
+#### String.new
+
+Since version 4.5.0
+
+Conversion to String is the most comprehensive conversion as there are many
+use cases where a String representation is wanted. The defaults for the many options
+have been chosen with care to be the most basic "value in textual form" representation.
+
+A new String can be created from all other data types. The process is performed in
+several steps - first the type of the given value is inferred, then the resulting type
+is used to find the most significant format specified for that type. And finally,
+the found format is used to convert the given value.
+
+The mapping from type to format is referred to as the format map. This map
+allows different formatting depending on type.
+
+Example Positive Integers in Hexadecimal prefixed with '0x', negative in Decimal
+
+~~~ puppet
+
+$format_map = { Integer[default, 0] => "%d",
+  Integer[1, default] => "%#x"
+}
+String("-1", $format_map)  # produces '-1'
+String("10", $format_map)  # produces '0xa'
+
+~~~
+
+A format is specified on the form:
+
+~~~
+
+%[Flags][Width][.Precision]Format
+
+~~~
+
+`Width` is the number of characters into which the value should be fitted. This allocated space is
+padded if value is shorter. By default it is space padded, and the flag 0 will cause padding with 0
+for numerical formats.
+
+`Precision` is the number of fractional digits to show for floating point, and the maximum characters
+included in a string format.
+
+Note that all data type supports the formats `s` and `p` with the meaning "default to string" and
+"default programmatic string representation" (which for example means that a String is quoted in 'p' format).
```

Comment:
"default string representation" and "default programmatic string representation" is more consistent than using "default to string".